crates/ide-diagnostics/src/tests.rs

```diff
@@ -143,3 +143,17 @@ fn test_disabled_diagnostics() {
     );
     assert!(!diagnostics.is_empty());
 }
+
+#[test]
+fn minicore_smoke_test() {
+    // Checks that there is no diagnostic in minicore for each flag.
+    for flag in MiniCore::available_flags() {
```

Comment:
Given flags can be nested to activate things only when multiple are active this doesn't make too much sense. I think it's fine to just check with all flags enabled.

Reply:
In this way it can catch missing dependencies between flags, which happened for me one time IIRC. But yes it seems it doesn't cover some cases. Is it fine to check one time for each flag and one time with all flags enabled? A complete check should try every 2^n subset of flags, which is definitely not worth it.

Reply:
I implemented the "one time for each flag and one time with all flags" approach.

Reply:
Doing both seems fine as well 👍